deployment/lib/lambdas/cf-url-rewriter.ts

```diff
@@ -0,0 +1,70 @@
+import { CloudFrontRequestEvent, CloudFrontRequest } from 'aws-lambda';
+
+const https = require('https');
+
+export async function handler(event: CloudFrontRequestEvent, context, callback): Promise<CloudFrontRequest> {
+  const request = event.Records[0].cf.request;
+
+  console.log('request', JSON.stringify(request));
+
+  if (request.uri.includes("latest")) {
+
+    const indexUri = request.uri.replace(/latest.*/, 'dist/index.json');
+
+    const data = await httpGet('https://' + request.headers.host[0].value + indexUri);
+
+    const redirectResponse = {
+      status: '302',
+      statusDescription: 'Moved temporarily',
+      headers: {
+        'location': [{
+          key: 'Location',
+          value: request.uri.replace('latest', data.latest),
+        }],
+        'cache-control': [{
+          key: 'Cache-Control',
+          value: "max-age=3600"
+        }],
+      },
+    };
+
+    console.log('update request', redirectResponse);
+
+    callback(null, redirectResponse);
+  } else {
+    request.uri = request.uri.replace(/^\/ci\/...\//, '\/');
+    callback(null, request);
+  }
+}
+
+async function httpGet(url) {
+  return new Promise((resolve, reject) => {
+
+    https.get(url, (res) => {
+      let body = "";
+
+      res.on("data", (chunk) => {
+        body += chunk;
+      });
+
+      res.on("end", () => {
+        try {
+          let json = JSON.parse(body);
+
+          console.log("json ", json);
+          resolve(json);
+
+          // do something with JSON
+        } catch (error) {
+          console.error(error.message);
+          reject(error);
+        };
+      });
+
+    }).on("error", (error) => {
```

Comment:
This function should catch all fail.

1. If the file exists, it succeeds.
2. If there's a 404, it's expected.
3. All other errors should fail visibly so that we're not guessing, or have to go dig in logs when the redirect is not working after, for example, setting the wrong permissions on the file.

Reply:
Added a try/catch logic at the caller part.